### ISSUE TITLE
libssh2 removes unrecognized entries in known_hosts

### ISSUE THREAD

**Issue by gnachman** [libssh2/libssh2#85]
I use this code to add an entry to known_hosts:
https://github.com/NMSSH/NMSSH/blob/master/NMSSH/NMSSHSession.m#L710

I received a report that entries using unrecognized algorithms are getting removed:
https://gitlab.com/gnachman/iterm2/issues/4250

I confirmed this by creating a known_hosts file like this (personal data removed):

````
foo.bar.com ssh-rsa *redacted*
example.com ecdsa-sha2-nistp256 *redacted*
````

I then added a new host, and the ecdsa-sha2-nistp256 entry was removed. Entries whose algorithm field is `ssh-rsa` do not get clobbered.

**Comment by alamaison** [comment]
What version of libssh2 are you linking against?  This was a known issue, but it was [fixed](https://github.com/libssh2/libssh2/commit/85c6627c861e970ba0bab10ec8b44b6523d6a5fd) 3 years ago.  The fix first appeared in 1.5.0.

I've added some more test cases based on your specific example, but they all pass.

**Comment by gnachman** [comment]
Sorry about that! I will make sure to update to the most recent version.
